val.py

```diff
@@ -441,36 +453,88 @@ def run(
                 save_one_json(predn, jdict, path, class_map)  # append to COCO-JSON dictionary
             callbacks.run('on_val_image_end', pred, predn, path, names, im[si])
 
-            # TODO the following code contains a bug and will be fixed in PR #13
-            pred_clone[:, :4] = scale_boxes(im.shape[2:], pred_clone[:, :4],
-                                            im0[si].shape).round()
+            if skip_evaluation or save_blurred_image:
+                # TODO the following code contains a bug and will be fixed in PR #13
+                predn[:, :4] = scale_boxes(im.shape[2:], predn[:, :4],
+                                           im0[si].shape).round()
+
+                image_filename, image_upload_date = parse_image_path(paths[si])
```

Comment:
I think parse_image_path should be moved out of this file, since when I run evaluation I don't need the path to be in date time format. What if we check the path structure in the settings of the inference/pre-inference pipeline?

in the settings schema we can add a check in the inference pipeline if the URI of the input attribute has the desired structure. that's what im doing in the evaluation pipeline. this way I don't need to worry and check inside val.py, since I'm already checking when I am parsing the config.yaml

Reply:
Changed the code that parse_image_path is only performed when the skip_evaluation flag is used. But let's have a call, your idea can make the code better.